rpc/client.go

```diff
@@ -236,6 +244,7 @@ func (rc *rpcClient) subscribeToNotifications(ctx context.Context, notificationC
 			case serde.ObjectiveCompleted:
 				rpcRequest := serde.JsonRpcSpecificRequest[protocols.ObjectiveId]{}
 				err := json.Unmarshal(data, &rpcRequest)
+				rc.logger.Debug("Received notification", "method", method, "data", rpcRequest)
```

Comment:
Instead of passing in the raw JSON we pass in a struct that will get serialized automatically for us.
```
{"time":"2023-08-22T11:43:47.327969-07:00","level":"INFO","msg":"Received notification","client":"aliceLedger","notification":{"ID":"0xe64cb6f532aaeb47a1b713280583494d6a6cfe0f6048df1cb0f10a8711e641c0","Status":"Proposed","Balance":{"AssetAddress":"0x0000000000000000000000000000000000000000","Me":"0x7e5f4552091a69125d5dfcb7b8c2659029395bdf","Them":"0x2b5ad5c4795c026514f8317c7a215e218dccd6cf","MyBalance":"0x64","TheirBalance":"0x64"}}}
```